pkg/injector/webhook.go

```diff
@@ -333,36 +334,34 @@ func patchMutatingWebhookConfiguration(cert certificate.Certificater, meshName,
 				ClientConfig: admissionv1beta1.WebhookClientConfig{
 					CABundle: cert.GetCertificateChain(),
 				},
-				Rules: []admissionv1beta1.RuleWithOperations{
-					{
-						Operations: []admissionv1beta1.OperationType{admissionv1beta1.Create},
-						Rule: admissionv1beta1.Rule{
-							APIGroups:   []string{""},
-							APIVersions: []string{"v1"},
-							Resources:   []string{"pods"},
-						},
-					},
-				},
 			},
 		},
 	}
-	data, err := json.Marshal(updatedWH)
+}
+
+// updateMutatingWebhookCABundle updates the existing MutatingWebhookConfiguration with the CA this OSM instance runs with.
+// It is necessary to perform this patch because the original MutatingWebhookConfig YAML does not contain the root certificate.
+func updateMutatingWebhookCABundle(cert certificate.Certificater, webhookName string, clientSet kubernetes.Interface) error {
```

Comment:
Maybe I missed this in the code somewhere but do we watch for changes in this mwhc resource to reconcile/ensure the CA cert in this mwhc is using the most recent CA cert from the secret? I can open an issue if that helps.

Reply:
@ritazh very good point - this does not exist *yet*!
An issue would be great!